### PR TITLE
perf:Optimize the status of READ_MEDIA_VISUAL_USER_SELECTED permission

### PR DESCRIPTION
**Pull request by inTao**
I adjusted the return of permission status; Added Android version version judgment

### CODE REVIEW

**Comment by mvanbeusekom** [comment]
Hi @inTao,

Would you mind updating the `version` in the `pubspec.yaml` file and add the change to the `CHANGELOG.md` (according to the conventions mentioned [here](https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog-style)) for the permission_handler_android package?

File: permission_handler_android/android/src/main/java/com/baseflow/permissionhandler/PermissionManager.java [hunk 1]
```diff
@@ -538,15 +538,17 @@ private int determinePermissionStatus(final @PermissionConstants.PermissionGroup
...
-                                permissionStatuses.add(PermissionUtils.determineDeniedVariant(activity, name));
-                            }
+                    final int permissionStatus = ContextCompat.checkSelfPermission(context, name);
+                    int permissionStatusLimited = permissionStatus;
+                    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.UPSIDE_DOWN_CAKE) {
+                        permissionStatusLimited = ContextCompat.checkSelfPermission(context, Manifest.permission.READ_MEDIA_VISUAL_USER_SELECTED);
+                    }
+                    if (permissionStatusLimited == PackageManager.PERMISSION_GRANTED && permissionStatus == PackageManager.PERMISSION_DENIED) {
+                        permissionStatuses.add(PermissionConstants.PERMISSION_STATUS_LIMITED);
+                    } else if (permissionStatusLimited == PackageManager.PERMISSION_GRANTED && permissionStatus == PackageManager.PERMISSION_GRANTED) {
```
Comment (LulleBulle) [inline permission_handler_android/android/src/main/java/com/baseflow/permissionhandler/PermissionManager.java:548]: is it necessary to check limited status here?

Comment (inTao) [reply permission_handler_android/android/src/main/java/com/baseflow/permissionhandler/PermissionManager.java]: You're right, I've deleted it

**Comment by LulleBulle** [comment]
Hello @inTao , have you run `flutter analyze` as per the contribution [guide](https://github.com/Baseflow/flutter-permission-handler/blob/main/CONTRIBUTING.md)? It seems to fail in the pipeline.

**Comment by inTao** [comment]
> Hello @inTao , have you run `flutter analyze` as per the contribution [guide](https://github.com/Baseflow/flutter-permission-handler/blob/main/CONTRIBUTING.md)? It seems to fail in the pipeline.

Can I just ignore it? I don't think it was caused by this change

**Comment by mvanbeusekom** [comment]
> > Hello @inTao , have you run `flutter analyze` as per the contribution [guide](https://github.com/Baseflow/flutter-permission-handler/blob/main/CONTRIBUTING.md)? It seems to fail in the pipeline.
> 
> Can I just ignore it? I don't think it was caused by this change 

No worries, I will fix them quickly before releasing these changes.

**Comment by mvanbeusekom** [comment]
@inTao, @LulleBulle, I have just released version 12.0.9 containing these changes. Thank you both for your helpful contributions.

**Pull request merged**